src/ring_theory/localization/as_subring.lean

```diff
@@ -0,0 +1,126 @@
+/-
+Copyright (c) 2022 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+import ring_theory.localization.localization_localization
+import tactic.field_simp
+
+/-!
+
+# Localizations of domains as subrings of the fraction field.
+
+Given a domain `A` with fraction field `K`, and a submonoid `S` of `A` which
+does not contain zero, this file constructs the localization of `A` at `S`
+as a subring of the field `K`.
+
+-/
+
+namespace localization
+
+variables
+  {A : Type*} (K : Type*) [comm_ring A] [is_domain A] [field K]
+  [algebra A K] [is_fraction_ring A K]
+
+open_locale non_zero_divisors
+
+lemma algebra_map_apply_ne_zero_of_le
+  {S : submonoid A} (hS : S ≤ A⁰) (s : A) (hs : s ∈ S) : algebra_map A K s ≠ 0 :=
+begin
+  intros c, rw ← (algebra_map A K).map_zero at c,
+  replace hs := hS hs, rw mem_non_zero_divisors_iff_ne_zero at hs,
+  replace c := is_fraction_ring.injective _ _ c,
+  exact hs c,
+end
+
+/--
+Given a domain `A` with fraction field `K`, and a submonoid `S` of `A` which
+contains no zero divisor, this is the localization of `A` at `S`, considered as
+a subring of `K`.
+-/
+def subring (S : submonoid A) (hS : S ≤ A⁰) : subring K :=
```

Comment:
Does it make sense to make this a subalgebra instead?

Reply:
Yes, absolutely! I'll do this soon.